src/app/lib/device/airplay.js

```diff
@@ -1,56 +1,119 @@
 (function (App) {
     'use strict';
 
-    var airplayer = require('airplayer'),
-        netw = require('network-address'),
-        list = airplayer(),
+    var Bonjour = require('bonjour'),
+        AirPlayProtocol = require('airplay-protocol');
+
+    var bonjourBrowser = new Bonjour(),
         collection = App.Device.Collection;
 
-    var makeID = function (baseID) {
-        return 'airplay-' + baseID.replace('.', '');
+    var makeID = function (bonjourService) {
+        // strong de-duplication ID since there can only be 1 bonjour service per host+port
+        var uniqueDeviceId = bonjourService.host + '_' + bonjourService.port;
+        return 'airplay-' + Common.md5(uniqueDeviceId);
     };
 
-    var Airplay = App.Device.Generic.extend({
+    var AirPlay = App.Device.Generic.extend({
         defaults: {
             type: 'airplay',
             typeFamily: 'external'
         },
-        makeID: makeID,
+
+        _makeID: makeID,
+
         initialize: function (attrs) {
             this.device = attrs.device;
-            this.attributes.id = this.makeID(this.device.host);
-            this.attributes.name = this.device.name || this.device.serverInfo.model;
-            this.attributes.address = netw();
+            this.attributes.id = this._makeID(this.device.service);
+            this.attributes.name = this.device.service.name;
+            this.attributes.address = this.device.service.host; // receiver's hostname, ie "AppleTV.local"
```

Comment:
I added this to prevent anyone from ever setting this to the local Butter Desktop IP again. ;p